server/fleet/host_certificates.go

```diff
@@ -139,8 +140,39 @@ type MDMAppleErrorChainItem struct {
 //
 // See https://osquery.io/schema/5.15.0/#certificates
 func ExtractDetailsFromOsqueryDistinguishedName(str string) (*HostCertificateNameDetails, error) {
-	// TODO
-	return nil, errors.New("not implemented")
+	str = strings.TrimSpace(str)
```

Comment:
It feels like there should be a built-in Go function to parse this format, but I couldn't find one (I'm not sure how that format is called). I found this: https://pkg.go.dev/crypto/x509/pkix#Name, but it takes a `RDNSequence` that is unmarshalled from `asn1` binary data, not something like the string osquery gives us. In any case, your implementation looks fine!

Reply:
Yeah, it took me some time trying to track down the provenance of the format used by osquery since it doesn't seem to be an official specification. Based on some online sleuthing, my best guess it that osquery is following "openSSL-style" formatting, which refers to the formatting requirements of the `-subj` flag for `openssl req`.